lib/net_mosq.c

```diff
@@ -34,7 +34,7 @@ and the Eclipse Distribution License is available at
 #include <sys/endian.h>
 #endif
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__NetBSD__)
```

Comment:
Possibly, these checks should go to the config.mk, and set something like, "HAS_NETINET_IN_H" instead, rather than modifying all the files individually?